hermes/probe/error/error.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove, GitHub: evanSpendlove.
+//
+// Error implements the ProbeError used in Hermes for recording the
+// exit status of functions.
+
+// Package error defines the ProbeError used in Hermes for recording the
+// exit status of functions.
+package error
+
+import (
+	"fmt"
+
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+)
+
+// ProbeError is an error that includes an exit status for used within Hermes.
+type ProbeError struct {
+	Status metrics.ExitStatus
```

Comment:
Nit: "Exit status" has a particular meaning: it's the numeric result code returned by an exiting process on Unix/Linux systems. Using the "ExitStatus" name suggests to the reader that there's a subprocess execution happening somewhere and we record the Unix exit status from the subprocess.
I would rename `ExitStatus` to `APICallStatus` or something similar to avoid this confusion.
It's not really important, so I'll leave it to you whether you do the rename.

Reply:
Okay, I wasn't aware of that. I've added it as a TODO, but I'd prefer not to make the change right now if possible.

Reply:
SGTM.